File: 0001a-theatre-square/solution.py
def solution():
    height, width, size = [int(x) for x in input().split()]
    height_ratio = height // size if height % size == 0 else height // size + 1
    width_ratio = width // size if width % size == 0 else width // size + 1
    area = height_ratio * width_ratio
    print(f"{area}")


if __name__ == "__main__":
    solution()

File: 0001a-theatre-square/solution.cpp
#include <iostream>


void solution() {
    unsigned long long height, width, size;
    std::cin >> height >> width >> size;

    unsigned long long height_ratio = (height % size == 0) ? (height / size) : ((height / size) + 1);
    unsigned long long width_ratio = (width % size == 0) ? (width / size) : ((width / size) + 1);
    std::cout << height_ratio * width_ratio << '\n';
}


void setup() {
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(NULL);
}


int main() {
    setup();
    solution();
}
